### ISSUE TITLE
ComboBox: Style keyboard focused list item with correct highlight

### ISSUE THREAD

**Issue by zepumph** [phetsims/sun#535]
While working on https://github.com/phetsims/sun/issues/462, I found that in ComboBoxListBox, the highlight listener is only for mouse, not for keyboard.

**Comment by zepumph** [comment]
@jessegreenberg please review.

**Comment by jessegreenberg** [comment]
This is a good improvement but has a potential problem because of https://github.com/phetsims/scenery/issues/1137. Since the ComboBox item receives focus when opened with a mouse, it looks selected when focus highlights are invisible. I think this is a problem because setting the highlight for one rectangle does not clear the other. The result can be the following, where focus is on the first item, mouse is over the sixth item.

![image](https://user-images.githubusercontent.com/6396244/114901680-090e4800-9de3-11eb-8c68-29b6d885412d.png)

If ComboBoxListItemNode used a PressListener, we could check PressListener.looksOverProperty.value before styling on focus (related to  https://github.com/phetsims/sun/issues/462).

Until then, I would recommend that we remove https://github.com/phetsims/sun/commit/be1aa17fe8a10dd2c775291c38a3c10bd9139601 and put this issue on hold until #462 is done. The focus highlight indicates the selected item well enough for now. @zepumph is that OK with you?

**Comment by zepumph** [comment]
Sounds good. On hold until https://github.com/phetsims/sun/issues/462 is done.

**Comment by marlitas** [comment]
Meeting 2/8:
- We looked at the current behavior.
- TS: This is a design question about consistency
- JG: This would be 2 effort points to change
- MK: Let's keep it open.

**Comment by zepumph** [comment]
Well that was quite easy! @jessegreenberg does this seem right to you? I looked at https://github.com/phetsims/sun/issues/462 and felt like it was appropriate to have our own, unattached PressListener. I was able to confirm that as nice behavior because I could get a highlight out of mouse, pdom focus, and touch all at the same time. Perhaps when we solve that other issue things could be combined to suit those needs. Anything else here?
